<launch>

  <!-- GDB functionality -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />
  <arg name="robot_name" default="tr1"/>

  <rosparam file="$(find tr1_hardware_interface)/config/hardware.yaml" command="load"/>
  <rosparam file="$(find tr1_hardware_interface)/config/controllers.yaml" command="load"/>
  <rosparam file="$(find tr1_hardware_interface)/config/joint_limits.yaml" command="load"/>
  <rosparam file="$(find tr1_hardware_interface)/config/joint_offsets.yaml" command="load"/>
	<param name="robot_description" command="cat $(find tr1_description)/urdf/tr1.urdf" />

  <node name="tr1_hardware_interface" pkg="tr1_hardware_interface" type="tr1_hardware_interface_node" output="screen" launch-prefix="$(arg launch_prefix)"/>
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/"
		args="
			/tr1/controller/state
			/tr1/controller/position/JointRightShoulderPan
			/tr1/controller/position/JointRightShoulderTilt
			/tr1/controller/position/JointRightUpperArmRoll
			/tr1/controller/position/JointRightElbowFlex
			/tr1/controller/position/JointRightForearmRoll
			/tr1/controller/position/JointRightWristFlex
			/tr1/controller/position/JointRightWristRoll
			/tr1/controller/effort/JointRightGripper
		"/>

</launch>
